lib/fs.js

```diff
@@ -1813,11 +1843,25 @@ fs.realpath = function realpath(path, options, callback) {
   } else if (typeof options !== 'object') {
     throw new TypeError('"options" must be a string or an object');
   }
-  callback = makeCallback(callback);
```

Comment:
The error thrown for `fs.realpath('\0')` changes from an asynchronous one to a synchronous one with this change, and error handling changes are usually considered semver-major. This one is certainly as edge-case-y as it gets, though.